### PR TITLE
MM-56781: Use correct error in doInviteRemote

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
A Sentry crash report identified this nil pointer dereference:
https://mattermost-mr.sentry.io/issues/4930233067/events/eae438652c7b4335be2bbe19c977f680

Interestingly, the stack trace shows the actual crash happening in the function deferred above this line, which makes sense, given that it's accessing the returned value, which is now nil as well. However, the origin of the crash is here, since it's using the previous `appErr`, which at this point is ensured to be nil, instead of the error returned by `InviteRemoteToChannel`. This makes the returned value that should have been generated by `responsef` to no longer exist and so the stack trace to point to that line as well.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56781

#### Screenshots
--

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — COMMENTED** [review]
Ugh, this is a very common mistake. Wondering if it's possible to add a test?

**Comment by agarciamontoro** [comment]
> Wondering if it's possible to add a test?

I tried for a bit, but there's a large amount of mocking there and ended up concluding the test wouldn't add much value. If you feel strongly about it, I can spend some time on it, but it seemed like busywork to me in this specific case.

I wonder whether a linter would be able to catch these things :thinking:

**Comment by agarciamontoro** [comment]
Ok, it was not that complex at the end of the day, but I want a double-check from @wiggin77 in two places, since I'm not familiar with this part of the codebase:
- Is the test accurate? I just wanted to stress the path where `InviteRemoteToChannel` fails, which is where the nil pointer dereferencing was happening in the first place. The test does stress that path, but I want another pair of eyes on it to make sure it makes sense.
- Does the new error make sense? I followed the structure of the other errors in the function, adding a more human-readable response, but I'm not sure if I'm missing something else.

File: server/channels/app/slashcommands/command_share_test.go [hunk 1]
```diff
@@ -90,4 +90,42 @@ func TestShareProviderDoCommand(t *testing.T) {
...
+		args := &model.CommandArgs{
+			T:         func(s string, args ...any) string { return s },
+			ChannelId: channel.Id,
+			UserId:    th.BasicUser.Id,
+			TeamId:    th.BasicTeam.Id,
+			Command:   "/share-channel invite --connectionID " + remoteCluster.RemoteId,
+		}
+
+		response := commandProvider.DoCommand(th.App, th.Context, args, "")
+		require.Contains(t, response.Text, args.T("api.command_share.invite_remote_to_channel.error"))
```
Comment (wiggin77) [inline server/channels/app/slashcommands/command_share_test.go:129]: What is meant to make this test fail?  It seems like the invite will succeed.

Comment (agnivade) [reply server/channels/app/slashcommands/command_share_test.go:129]: @agarciamontoro - You can just do `require.NotPanics` if you don't have anything to verify. But if you do, that is fine as well.

**Review by agnivade — APPROVED** [review]

File: server/channels/app/slashcommands/command_share_test.go (hunk 1 above)
Comment (wiggin77) [reply server/channels/app/slashcommands/command_share_test.go:129]: Ah, I forgot that `th.CreateChannel` will create the channel with a bogus remoteID, thus making it appear to originate somewhere else.  Test looks good, and thanks for fixing this.

File: server/channels/app/slashcommands/command_share_test.go [hunk 2]
```diff
@@ -90,4 +90,42 @@ func TestShareProviderDoCommand(t *testing.T) {
...
+			Token:      model.NewId(),
+			Topics:     "topic",
+			CreateAt:   model.GetMillis(),
+			LastPingAt: model.GetMillis(),
+			CreatorId:  model.NewId(),
+		})
+		require.Nil(t, err)
+
+		commandProvider := ShareProvider{}
+		channel := th.CreateChannel(th.BasicTeam, WithShared(true))
```
Comment (wiggin77) [inline server/channels/app/slashcommands/command_share_test.go:119]:
```suggestion
		channel := th.CreateChannel(th.BasicTeam, WithShared(true))  // will create with generated remoteID
```

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**